File: app/src/cpp/CSPrefs.cpp
//
// Created by chris on 1/9/17.
//
#include <util/stl.hpp>
#include <jni.h>
#include <glitch-common.h>
#include <string>
#include "JavaClass.h"
#include "CSPrefs.h"


CSPrefs::CSPrefs(CSApplication &app) : preferencesClass(*app.sharedPrefsClass), editorClass(*app.sharedPrefsEditorClass){
    JavaClass preferenceManagerClass(app, "android/preference/PreferenceManager");
    jmethodID p = preferenceManagerClass.getStaticMethod(app.getEnv(), "getDefaultSharedPreferences", &preferencesClass, app.contextClass);
    sharedPrefs = app->NewGlobalRef(app->CallStaticObjectMethod(preferenceManagerClass.getClass(), p, /*app.mainActivityObject*/app.applicationContext));

    editMethod = preferencesClass.getMethod(app.getEnv(), "edit", &editorClass);
    getIntMethod = preferencesClass.getMethod(app.getEnv(), "getInt", 'I', app.stringClass, 'I');
    putIntMethod = editorClass.getMethod(app.getEnv(), "putInt", &editorClass, app.stringClass, 'I');
    applyMethod = editorClass.getMethod(app.getEnv(), "apply", 'V');
}

int CSPrefs::getInt(CSApplication& app, const char *key) {
    jstring jkey = app->NewStringUTF(key);
    return app->CallIntMethod(sharedPrefs, getIntMethod, jkey, -1);
}

void CSPrefs::putInt(CSApplication &app, const char *key, int val) {
    jobject editor = app->CallObjectMethod(sharedPrefs, editMethod);
    app->CallObjectMethod(editor, putIntMethod, app->NewStringUTF(key), val);
    app->CallVoidMethod(editor, applyMethod);
}

File: app/java_backup/com/example/chris/imagebender/CSPrefs.java
package com.example.chris.imagebender;

import android.app.Activity;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by chris on 12/18/16.
 */

public class CSPrefs {
    private SharedPreferences prefs;
    CSPrefs(Activity mainActivity) {
        prefs = PreferenceManager.getDefaultSharedPreferences(mainActivity);
    }

    void putInt(String key, int value) {
        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(key, value);
        editor.apply();
    }
    //our default value is always -1
    int getInt(String key) {
        return prefs.getInt(key, -1);
    }
}
